seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/sql/zeta/ZetaSQLFilter.java

```diff
@@ -162,11 +170,29 @@ private boolean likeExpr(LikeExpression likeExpression, Object[] inputFields) {
         }
         Expression rightExpr = likeExpression.getRightExpression();
         Object rightVal = zetaSQLFunction.computeForValue(rightExpr, inputFields);
-        if (rightVal == null) {
+        String regex = rightVal.toString();
+        if (rightVal == null && regex.length() > 0) {
             return false;
         }
-
-        String regex = rightVal.toString().replace("%", ".*").replace("_", ".");
+        String likeIdent = "%";
+        if (regex.startsWith(likeIdent)) {
+            regex = regex.replaceFirst(likeIdent, ".*");
+        }
+        if (regex.endsWith(likeIdent)) {
+            regex = regex.substring(0, regex.length() - 1) + ".*";
+        }
+        if (regex.startsWith("_")) {
+            regex = regex.replaceFirst("_", ".");
+        }
+        if (regex.endsWith("_")) {
+            regex = regex.substring(0, regex.length() - 1) + ".";
+        }
+        if (regex.length() >= 3 && regex.substring(regex.length() - 3).endsWith("_.*")) {
+            regex = regex.substring(0, regex.length() - 3) + "..*";
+        }
+        if (regex.startsWith("'") && regex.endsWith("'")) {
+            regex = regex.substring(0, regex.length() - 1).substring(1);
+        }
         if (regex.startsWith("'") && regex.endsWith("'")) {
             regex = regex.substring(0, regex.length() - 1).substring(1);
         }
```

Comment:
Why this logic exist twice?

Reply:
sorry,My question, thank you for your review